src/wallet.js

```diff
@@ -502,13 +502,14 @@ const wallet = {
    * @param {number} count How many addresses I will load
    * @param {Connection} connection Connection object to subscribe for the addresses
    * @param {Store} store Store object to save the data
+   * @param {string[]} preCalculatedAddresses  An array of pre-calculated addresses for this wallet
    *
    * @return {Promise} Promise that resolves when addresses history is finished loading from server
    *
    * @memberof Wallet
    * @inner
    */
-  loadAddressHistory(startIndex, count, connection = null, store = null) {
+  loadAddressHistory(startIndex, count, connection = null, store = null, preCalculatedAddresses = null) {
```

Comment:
We have tests for the method `executeGenerateWallet`, which calls `loadAddressHistory`. However, there is no way to execute `loadAddressHistory` with `preCalculatedAddresses !== null` with `loadAddressHistory`.

That's why I think you could create a test for this method using `preCalculatedAddresses` parameter.

Reply:
Good idea!
Created a new test for the `loadAddressHistory` method on [e52e1f0](https://github.com/HathorNetwork/hathor-wallet-lib/pull/354/commits/e52e1f0d06b05981666c2babf9cabc8178141892).